preset/vault/preset.go

```diff
@@ -0,0 +1,162 @@
+// interface. This Preset can be passed to gnomock.Start() function to create a
+// configured vault container to use in tests.
+package vault
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/orlangure/gnomock"
+	"github.com/orlangure/gnomock/internal/registry"
+)
+
+const (
+	defaultVersion   = "1.13.0"
+	defaultPort      = 8200
+	defaultAuthToken = "gnomock-vault-token" //nolint:gosec
+)
+
+func init() {
+	registry.Register("vault", func() gnomock.Preset { return &P{} })
+}
+
+// Preset creates a new Gmomock Vault preset. This preset includes a
+// vault specific healthcheck function and default vault image and port.
+func Preset(opts ...Option) gnomock.Preset {
+	p := &P{}
+
+	for _, opt := range opts {
+		opt(p)
+	}
+
+	return p
+}
+
+// P is a Gnomock Preset implementation for vault.
+type P struct {
+	Version     string       `json:"version"`
+	AuthToken   string       `json:"auth_token"`
+	Auth        []Auth       `json:"auth"`
+	Policies    []Policy     `json:"policies"`
+	TokenCreate *TokenCreate `json:"token_create"`
+}
+
+// Auth represents a vault authorization.
+type Auth struct {
+	Path string `json:"path"`
+	Type string `json:"type"`
+}
+
+// Policy is a vault policy.
+type Policy struct {
+	Name string `json:"name"`
+	Data string `json:"data"`
+}
+
+// TokenCreate creates a token with the providend policies in the provided file path.
+type TokenCreate struct {
+	Policies []string `json:"policies"`
+	FilePath string   `json:"file_path"`
+}
+
+// Image returns an image that should be pulled to create this container.
+func (p *P) Image() string {
+	return fmt.Sprintf("hashicorp/vault:%s", p.Version)
+}
+
+// Ports returns ports that should be used to access this container.
+func (p *P) Ports() gnomock.NamedPorts {
+	return gnomock.DefaultTCP(defaultPort)
+}
+
+// Options returns a list of options to configure this container.
+func (p *P) Options() []gnomock.Option {
+	p.setDefaults()
+
+	opts := []gnomock.Option{
+		gnomock.WithEnv("VAULT_DEV_ROOT_TOKEN_ID=" + p.AuthToken),
+		gnomock.WithInit(p.initf()),
+		gnomock.WithHealthCheck(p.healthcheck),
+	}
+
+	return opts
+}
+
+func (p *P) setDefaults() {
+	if p.Version == "" {
+		p.Version = defaultVersion
+	}
+
+	if p.AuthToken == "" {
+		p.AuthToken = defaultAuthToken
+	}
+}
+
+func (p *P) initf() gnomock.InitFunc {
+	return func(ctx context.Context, c *gnomock.Container) error {
+		cli, err := p.client(c)
+		if err != nil {
+			return err
+		}
+
+		for _, m := range p.Auth {
+			if err := cli.Sys().EnableAuthWithOptions(m.Path, &api.EnableAuthOptions{
+				Type:        m.Type,
+				Description: m.Path,
+			}); err != nil {
+				return fmt.Errorf("failed to enable %s auth on %s: %w", m.Type, m.Path, err)
+			}
+		}
+
+		for _, pol := range p.Policies {
+			if err := cli.Sys().PutPolicy(pol.Name, pol.Data); err != nil {
+				return fmt.Errorf("failed to write policy %s: %w", pol.Name, err)
+			}
+		}
+
+		if p.TokenCreate != nil {
```

Comment:
I'm not sure this option fits here. Generally preset options modify the container, but this one actually makes changes to the client file system.

I do like the idea of helper functions though. There is [`Ingest`](https://github.com/orlangure/gnomock/blob/97f7ed988cf52e93f58ae19df633512c37842bdb/preset/splunk/init.go#L92) function in Splunk preset for example. It does not modify the client, but it also doesn't belong in the options. What do you think about removing it as an option, but creating a helper function instead that can be called by clients (test code) whenever a new token needs to be configured?

Reply:
I agree with you. I removed the option and created two helper methods which can be handy in tests:

```golang
// Client creates a configured vault client for the provided container and token.
func Client(c *gnomock.Container, token string) (*api.Client, error) {
...
}
```

and 

```golang
// CreateToken creates an additional access token with the provided policies. Use the same password you provided
// with the WithAuthToken option.
func CreateToken(c *gnomock.Container, rootToken string, policies ...string) (string, error) {
...
}
```

If all is ok, I can squash the commits and update the MR if you want.